src/main/java/fruitshop/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,20 @@
+package fruitshop.strategy;
+
+import fruitshop.db.Storage;
+import fruitshop.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    private final Storage storage = new Storage();
```

Comment:
The `Storage` instance should not be created inside the `PurchaseOperation` class. This violates the Single Responsibility Principle (SRP) and the Dependency Inversion Principle (DIP) of SOLID. Instead, `Storage` should be passed as a dependency to the `PurchaseOperation` constructor.